### PR TITLE
network-ng: fix autoyast dns handling

### PR DESCRIPTION
**Pull request by imobachgs**

### CODE REVIEW

File: src/lib/y2network/hostname_reader.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+      host, _domain = *Yast::Hostname.SplitFQ(fqdn)
+      host.empty? ? nil : host
+    end
+
+    # Reads the hostname from the underlying system
+    #
+    # @return [String] Hostname
+    def hostname_from_system
+      Yast::Execute.on_target!("/bin/hostname", "--fqdn", stdout: :capture)
```
Comment (jreidinger) [inline src/lib/y2network/hostname_reader.rb]: I think now prefered is to run it as `/usr/bin/hostname`

File: src/lib/y2network/autoinst/dns_reader.rb [hunk 2]
```diff
@@ -40,9 +41,9 @@ def initialize(section)
       def config
         Y2Network::DNS.new(
           dhcp_hostname:      section.dhcp_hostname,
-          hostname:           section.hostname,
+          hostname:           section.hostname || default_hostname,
           nameservers:        valid_ips(section.nameservers),
-          resolv_conf_policy: section.resolv_conf_policy,
+          resolv_conf_policy: section.resolv_conf_policy || "auto",
```
Comment (kobliha) [inline src/lib/y2network/autoinst/dns_reader.rb]: Ideally not a string, but a constant (and used elsewhere too)

File: src/lib/y2network/hostname_reader.rb [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    def hostname_from_system
+      Yast::Execute.on_target!("/usr/bin/hostname", "--fqdn", stdout: :capture).strip
+    rescue Cheetah::ExecutionFailed
+      name = Yast::SCR.Read(Yast::Path.new(".target.string"), "/etc/hostname").to_s.strip
+      name.empty? ? nil : name
+    end
+
+    # @return [Array<String>] Valid chars to be used in the random part of a hostname
+    HOSTNAME_CHARS = (("a".."z").to_a + ("0".."9").to_a).freeze
+    private_constant :HOSTNAME_CHARS
```
Comment (kobliha) [inline src/lib/y2network/hostname_reader.rb]: Looks like Punycode is not supported, see https://en.wikipedia.org/wiki/Punycode or https://es.wikipedia.org/wiki/Punycode

Comment (kobliha) [reply src/lib/y2network/hostname_reader.rb]: Hmm, taking back, this seems to be used just for random hostname, not for validation...

Comment (imobachgs) [reply src/lib/y2network/hostname_reader.rb]: These characters are only used to generate a random hostname (`linux-ab3d`, for instance).

File: src/lib/y2network/autoinst/dns_reader.rb (hunk 2 above)
Comment (imobachgs) [reply src/lib/y2network/autoinst/dns_reader.rb]: Yes, actually we might think about having an enum, but it is basically only used here. I will move it to a constant though.

**Pull request merged**